arbnode/node.go

```diff
@@ -218,7 +218,9 @@ func GenerateRollupConfig(prod bool, wasmModuleRoot common.Hash, rollupOwner com
 		WasmModuleRoot:           wasmModuleRoot,
 		Owner:                    rollupOwner,
 		LoserStakeEscrow:         loserStakeEscrow,
-		ChainId:                  chainId,
+		ChainId:                  chainConfig.ChainID,
+		// TODO could the ChainConfig be just []byte?
+		ChainConfig: string(serializedChainConfig),
```

Comment:
in `deploy.go` serializedChainConfig is loaded from file - it is unmarshaled for sake of validation, but the unchanged json file content is passed to `GenerateRollupConfig`.

The question I had was about boundary between Solidity and Go - is there need for converting chain config `[]byte` to `string` or could rollup config contain bytes array instead of string?

Reply:
The two types are pretty much equivalent, I don't think there's any risk in casting between them